generator/internal/codegen_utils.h

```diff
@@ -63,6 +65,21 @@ std::string ServiceNameToFilePath(absl::string_view service_name);
  */
 std::string ProtoNameToCppName(absl::string_view proto_name);
 
+enum class NamespaceType { NORMAL, INTERNAL };
+
+/**
+ * Builds namespace hierarchy.
+ */
+std::vector<std::string> BuildNamespaces(
+    std::map<std::string, std::string> const& vars,
+    NamespaceType ns_type = NamespaceType::NORMAL);
+
+/**
+ * Validates command line arguments passed to the microgenerator.
```

Comment:
Can you document what the `parameters` string must look like? I'd guess a string of `"key=val foo=bar blah=baz"`, but that's just a guess.